arbitrator/prover/src/merkle.rs

```diff
@@ -134,7 +108,7 @@ impl MerkleType {
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
 struct Layers {
     data: Vec<Vec<Bytes32>>,
```

Comment:
Also probably better for a follow-up PR, but I wonder if `data: Vec<u8>` might perform better, because when hashing the layer we could take a slice of the two nodes together (and have another branch if there's only one node and an empty node).

Reply:
I added this task to a comment on the linear ticket: https://linear.app/offchain-labs/issue/NIT-2411/arbitrator-optimizations-to-reduce-number-of-challenge-levels-to-3#comment-86e91b86